src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -37,24 +42,8 @@ default boolean isSameStateAs(ReadOnlyPerson other) {
      */
     default String getAsTextShowAll() {
         final StringBuilder builder = new StringBuilder();
-        final String detailIsPrivate = "(private) ";
-        builder.append(getName())
-                .append(" Phone: ");
-        if (getPhone().isPrivate()) {
-            builder.append(detailIsPrivate);
-        }
-        builder.append(getPhone())
-                .append(" Email: ");
-        if (getEmail().isPrivate()) {
-            builder.append(detailIsPrivate);
-        }
-        builder.append(getEmail())
-                .append(" Address: ");
-        if (getAddress().isPrivate()) {
-            builder.append(detailIsPrivate);
-        }
-        builder.append(getAddress())
-                .append(" Tags: ");
+        builder.append(getPrintableString(getName(), getPhone(), getEmail(), getAddress()));
```

Comment:
A better way to do this is to implement a method call  getPrintableString(Printable... printables) and concatenate the string inside the method and you call that method here instead. Don't you think so?